modules/document_helper.py

```diff
@@ -0,0 +1,122 @@
+import re
+
+from gensim.parsing.preprocessing import remove_stopwords
+import lxml.etree as ET
+import pandas as pd
+
+
+filter_tag_list = [
+    "sc", "italic", 
+    "xref", "label", 
+    "sub", "sup", 
+    "inline-formula", "fig", 
+    "disp-formula", "bold", 
+    "table-wrap", "table", 
+    "thead", "tbody",
+    "caption", "tr", "td"
+]
+
+def dump_article_text(file_path, xpath_str, filter_tags=filter_tag_list, remove_stop_words=True):
+    """
+    This method is designed to extract all text from xml documents. 
+    Every document has specific tags that are striped in order to produce
+    clean text output for downstream processing
+    
+    Keyword arguments:
+        file_path - the file path for xml document
+        xpath_str - the xpath string to extract tags from the xml document
+        filter_tag_list - the list of tags to strip from the xml document
+        remove_stop_words - a flag to indicate if stop words should be removed
+    """
+    
+    parser = ET.XMLParser(encoding='UTF-8', recover=True)
+    tree = (
+        ET.parse(
+            open(file_path, "rb"),
+            parser = parser
+        )
+    )
+    
+    # Process xml without specified tags
+    ET.strip_tags(tree, *filter_tags)
+
+    root = tree.getroot()
+    all_tags = root.xpath(xpath_str)
+    text = list(map(lambda x: list(x.itertext()), list(all_tags)))
+    
+    # Remove stop words
+    if remove_stop_words:
+        text = (
+            list(
+                map(
+                    lambda x: remove_stopwords(re.sub("\n", "", "".join(x))),
+                    text
+                )
+            )
+        )
+
+    else:
+        text = (
+            list(
+                map(
+                    lambda x: re.sub("\n", "", "".join(x)), 
+                    text
+                )
+            )
+        )
+
+    return text
+
+def generate_doc_vector(model, document_path, xpath, filter_tags=filter_tag_list):
+    """
+    This method is designed to construct document vectors for a given xml document. 
+    Every document has specific tags that are striped in order to have accurate embeddings
+    
+    Keyword arguments:
+        model - the model to extract word vectors from
+        xpath_str - the xpath string to extract tags from the xml document
+        filter_tag_list - the list of tags to strip from the xml document
+    """
+    
+    parser = ET.XMLParser(encoding='UTF-8', recover=True)
```

Comment:
I'm a little confused. I assumed `dump_article_text` was parsing your article=document and then `generate_doc_vector` took that parsed document (i.e list of words in the document) and embedded it into a vector (i.e. number of occurrences of words)?

Does it make sense to have a separate parser function if being used in two different functions?

Reply:
Good catch. Since parser is being used in both functions it would make more sense to have it be a global variable. I'll make that change.

You are also correct. `dump_article_text` takes in a document and outputs extracted text.  `generate_doc_vector` takes in a document and outputs a vector corresponding to an average of all words within a document.